docs/csharp/language-reference/statements/declarations.md

```diff
@@ -56,90 +56,62 @@ When you work with [anonymous types](../../fundamentals/types/anonymous-types.md
 
 ```csharp
-public ref Person GetContactInformation(string fname, string lname)
+ref int alias = ref variable;
 ```
 
-Let's contrast these two assignments:
+A reference variable is a variable that refers to another variable, which is called the *referent*. That is, a reference variable is an *alias* to its referent. When you assign a value to a reference variable, that value is assigned to the referent. When you read the value of a reference variable, the referent's value is returned. The following example demonstrates that behavior:
 
-```csharp
-Person p = contacts.GetContactInformation("Brandie", "Best");
-ref Person p2 = ref contacts.GetContactInformation("Brandie", "Best");
-```
-
-The variable `p` holds a *copy* of the return value from `GetContactInformation`. It's a separate storage location from the `ref` return from `GetContactInformation`. If you change any property of `p`, you are changing a copy of the `Person`.
-
-The variable `p2` *refers to* the storage location for the `ref` return from `GetContactInformation`. It's the same storage as the `ref` return from `GetContactInformation`. If you change any property of `p2`, you are changing that single instance of a `Person`.
-
-You can access a value by reference in the same way. In some cases, accessing a value by reference increases performance by avoiding a potentially expensive copy operation. For example, the following statement shows how one can define a ref local value that is used to reference a value.
-
-```csharp
-ref VeryLargeStruct reflocal = ref veryLargeStruct;
-```
-
-The `ref` keyword is used both before the local variable declaration *and* before the value in the second example. Failure to include both `ref` keywords in the variable declaration and assignment in both examples results in compiler error CS8172, "Can't initialize a by-reference variable with a value."
+:::code language="csharp" source="snippets/declarations/ReferenceVariables.cs" id="AliasToLocalVariable":::
 
-```csharp
-ref VeryLargeStruct reflocal = ref veryLargeStruct; // initialization
-refLocal = ref anotherVeryLargeStruct; // reassigned, refLocal refers to different storage.
-```
-
-Ref local variables must still be initialized when they're declared.
+Use the [`ref` assignment operator](../operators/assignment-operator.md#ref-assignment) `= ref` to change the referent of a reference variable, as the following example shows:
 
-The following example defines a `NumberStore` class that stores an array of integer values. The `FindNumber` method returns by reference the first number that is greater than or equal to the number passed as an argument. If no number is greater than or equal to the argument, the method returns the number in index 0.
+:::code language="csharp" interactive="try-dotnet-method" source="snippets/declarations/ReferenceVariables.cs" id="RefReassign":::
```

Comment:
Sadly, this example doesn't work in C# interactive (the index operator isn't supported:

```suggestion
:::code language="csharp" source="snippets/declarations/ReferenceVariables.cs" id="RefReassign":::
```

Reply:
Ah, thank you! I've intended to make the very first snippet interactive. Fixed it now. Thank you for the catch.